include/boundary_region.hxx

```diff
@@ -15,13 +15,20 @@ namespace globals {
 } // namespace globals
 
 /// Location of boundary
-enum BndryLoc {BNDRY_XIN=1,
-               BNDRY_XOUT=2,
-               BNDRY_YDOWN=4,
-               BNDRY_YUP=8,
-               BNDRY_ALL=15,
-               BNDRY_PAR_FWD=16,   // Don't include parallel boundaries
-               BNDRY_PAR_BKWD=32};
+enum class BndryLoc {xin=1,
+                     xout=2,
+                     ydown=4,
+                     yup=8,
+                     all=15,
+                     par_fwd=16,   // Don't include parallel boundaries
+                     par_bkwd=32};
```

Comment:
I just noticed the numbering for `BndryLoc` seems to be set up for some kind of binary-flag operations. An `enum class` probably doesn't work if we need to be able to pass an argument `3` and have it be interpreted as `xin + xout`. However, in `field_data.cxx` the `BndryLoc` is used directly as a key of a `std::map`, so I guess we don't currently support combining `BndryLoc`s. Do we want to? If so, should I revert the `BndryLoc` change?

Reply:
I think this was the original intention, but was never (?) used that way. It would be nice if there were a way to compose boundaries, but there is probably a better way than bit twiddling. 

Presumably the change would catch any code attempting to add `enum class`es, so if nothing turns up then I'm happy for this change to be kept. A better/different way of adding boundaries can be implemented later...

Reply:
Looks like it's also used in `switch`es elsewhere too, and there's no bit-twiddling going on that I can see.

I would be worried about changing the numbers, in case someone is using the integer values to convert to the `enum`.

Reply:
@ZedThree If anyone's passing `ints` and having them be implicitly converted, wouldn't the change to `enum class` break that use anyway? I doubt anyone's doing that, but if they are they'd have to change like `somefunction(2)` to `somefunction(<BndryLoc> 2)` even if we didn't get rid of the explicit values, and they might as well change to `BNDRY_XOUT` or `BndryLoc::xout` at the same time?

Reply:
BTW, what does the `// Don't include parallel boundaries` comment mean? If `par_fwd` or `par_bkwd` is being set, isn't that including a 'parallel boundary'?